src/js/step.js

```diff
@@ -184,35 +183,20 @@ export class Step extends Evented {
   }
 
   /**
-   * Attaches final element to default or passed location
+   * Creates Shepherd element for step based on options
    *
    * @private
-   * @param {HTMLElement} element The element to attach
+   * @return {HTMLElement} The DOM element for the step tooltip
    */
-  _attach(element) {
-    const { renderLocation } = this.options;
+  _createTooltipContent() {
+    const classes = this.options.classes || '';
+    const element = createFromHTML(`<div class='${classes}' data-id='${this.id}'>`);
 
-    if (renderLocation) {
-      if (renderLocation instanceof HTMLElement) {
-        return renderLocation.appendChild(element);
-      }
-      if (isString(renderLocation)) {
-        return document.querySelector(renderLocation).appendChild(element);
-      }
+    if (this.options.idAttribute) {
```

Comment:
What is this `idAttribute` stuff?

Reply:
ID has been null in recent versions, does this fix that?

Reply:
From what I'm seeing now, no. Gets set in a Step, but I don't see where in options this is passed or has a fallback.

Reply:
Yeah, I was seeing ids of `undefined` on tooltips when `options.idAttribute` didn't exist. I'm not sure why the diff show it against removing `renderLocation`, but the original code was doing:

```js
    const element = createFromHTML(`<div class='${classes}' data-id='${this.id}' id='${this.options.idAttribute}>`);
```

Reply:
Admittedly, it's a bit of a random change. I'm definitely in favor or using https://github.com/shipshapecode/shepherd/pull/262 instead.